From: Michael Chan <mchan@broadcom.com>
Date: Wed, 14 Nov 2012 14:44:29 +0000
Subject: [PATCH 096/105] tg3: Use tp->rxq_cnt when checking RSS tables.
Patch-mainline: v3.8-rc1
Git-commit: 0b3ba0553a9ef578dd289d9eed65cbd3b4823211
References: bnc#790588 FATE#313912

irq_cnt is no longer reliable since rxq_cnt can be independently configured.
Update version to 3.127.

Signed-off-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Nithin Nayak Sujir <nsujir@broadcom.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/broadcom/tg3.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -8673,14 +8673,14 @@ static void tg3_rss_check_indir_tbl(stru
 	if (!tg3_flag(tp, SUPPORT_MSIX))
 		return;
 
-	if (tp->irq_cnt <= 2) {
+	if (tp->rxq_cnt == 1) {
 		memset(&tp->rss_ind_tbl[0], 0, sizeof(tp->rss_ind_tbl));
 		return;
 	}
 
 	/* Validate table against current IRQ count */
 	for (i = 0; i < TG3_RSS_INDIR_TBL_SIZE; i++) {
-		if (tp->rss_ind_tbl[i] >= tp->irq_cnt - 1)
+		if (tp->rss_ind_tbl[i] >= tp->rxq_cnt)
 			break;
 	}
 
